analyzers/src/SonarAnalyzer.Common/Rules/StringLiteralShouldNotBeDuplicatedBase.cs

```diff
@@ -87,10 +87,17 @@ private void ReportOnViolation(SyntaxNodeAnalysisContext context)
             {
                 var duplicates = item.ToList();
                 var firstToken = duplicates[0];
+                var messageText = ExtractStringContent(firstToken);
                 context.ReportIssue(Diagnostic.Create(rule, firstToken.GetLocation(),
                     duplicates.Skip(1).Select(x => x.GetLocation()),
-                    item.Key, duplicates.Count));
+                    messageText, duplicates.Count));
```

Comment:
`messageText` is somehow confusing, as it's not the message text, but the argument holding the string content. I'd remove the variable and call `ExtractStringContent` directly there.